sbpy/photometry/core.py

```diff
@@ -99,24 +104,306 @@ def mag2ref(mag, radius, M_sun=None):
+
+        .. math::
+
+            α = R(λ1) / R(λ0) = 10^{0.4 color_index}
+
+            color_index = Δm - C_{sun}
+
+        Δλ is typically expressed in units of 100 nm.
+
+
+        Examples
+        --------
+        >>> import astropy.units as u
+        >>> w = [0.4719, 0.6185] * u.um
+        >>> S = SpectralGradient.from_color(w, 0.10 * u.mag)
+        >>> print(S)                            # doctest: +FLOAT_CMP
+        6.27819572 % / 100 nm
+
+        """
+
+        eff_wave = SpectralGradient._eff_wave(wfb)
+
+        try:
+            # works for u.Magnitudes and dimensionless u.Quantity
+            alpha = u.Quantity(color, u.dimensionless_unscaled)
+        except u.UnitConversionError:
+            # works for u.mag
+            alpha = color.to(u.dimensionless_unscaled, u.logarithmic())
+
+        dw = eff_wave[0] - eff_wave[1]
+        S = ((2 / dw * (alpha - 1) / (alpha + 1))
+             .to(u.percent / hundred_nm))
+
+        return SpectralGradient(S, wave=eff_wave)
+
+    def to_color(self, wfb):
+        r"""Express as a color index.
+
+
+        Parameters
+        ----------
+        wfb : two-element `~astropy.units.Quantity` or tuple
+            Wavelengths, frequencies, or bandpasses of the
+            measurement.  If a bandpass, the effective wavelength of a
+            solar spectrum will be used.  Bandpasses may be a string
+            (name) or `~synphot.SpectralElement` (see
+            :func:`~sbpy.spectroscopy.sun.Sun.filt`).
+
+        Notes
+        -----
+        Color index is computed from:
+
+        .. math::
+
+            α = \frac{1 + S Δλ / 2}{1 - S * Δλ / 2}
+
+        where S is the spectral gradient at the mean of λ0 and λ1, and:
+
+        .. math::
+
+            α = R(λ1) / R(λ0) = 10^{0.4 color_index}
+
+            color_index = Δm - C_{sun}
+
+        Δλ is typically expressed in units of 100 nm.
+
+
+        Returns
+        -------
+        color : `~astropy.units.Quantity`
+            ``blue - red`` color in magnitudes, dimensionless and
+            excludes the solar color.
+
+
+        Examples
+        --------
+        >>> import astropy.units as u
+        >>> from sbpy.units import hundred_nm
+        >>> S = SpectralGradient(10 * u.percent / hundred_nm,
+        ...                      wave0=0.55 * u.um)
+        >>> C = S.to_color((525, 575) * u.nm)
+        >>> print(C)    # doctest: +FLOAT_CMP
+        0.05429812423309064 mag
+
+        """
+
+        eff_wave = self._eff_wave(wfb)
+
+        S = self.renormalize(eff_wave.mean())
+        dw = eff_wave[0] - eff_wave[1]
+        beta = (S * dw / 2).decompose()  # dimensionless
+        color = ((1 + beta) / (1 - beta)).to(u.mag, u.logarithmic())
+
+        return color
+
+    def renormalize(self, wave0):
+        """Re-normalize to another wavelength.
+
+        The slope is linearly extrapolated to the new normalization
+        point.  Requires the `wave` attribute to be defined, see
```

Comment:
`wave0` is required, but `wave` isn't, right?

Reply:
Correct, will fix that too.